### PR TITLE
Vulkan validation fixes

### PR DESCRIPTION
**Pull request by hrydgard**
Now we mostly pass Vulkan 1.0.61 validation layer checks.

### CODE REVIEW

File: Common/Vulkan/VulkanMemory.cpp [hunk 1]
```diff
@@ -47,10 +47,16 @@ bool VulkanPushBuffer::AddBuffer() {
...
 	}
 
+	VkMemoryRequirements reqs;
+	vkGetBufferMemoryRequirements(device_, info.buffer, &reqs);
+
+	// TODO: We really should use memoryTypeIndex here..
+
 	// Okay, that's the buffer. Now let's allocate some memory for it.
 	VkMemoryAllocateInfo alloc = { VK_STRUCTURE_TYPE_MEMORY_ALLOCATE_INFO };
+	// TODO: Should check here that memoryTypeIndex_ matches reqs.memoryTypeBits.
```
Comment (unknownbrackets) [inline Common/Vulkan/VulkanMemory.cpp]: Any reason not to do it like here?

https://github.com/hrydgard/ppsspp/blob/804aa79376eb4cb216f9f7fa1b8015eff1d3ac59/Common/Vulkan/VulkanMemory.cpp#L162

We could just basically do this later, I think:

https://github.com/hrydgard/ppsspp/blob/804aa79376eb4cb216f9f7fa1b8015eff1d3ac59/Common/Vulkan/VulkanMemory.cpp#L24

-[Unknown]

File: GPU/Vulkan/DrawEngineVulkan.cpp [hunk 2]
```diff
@@ -884,8 +884,11 @@ void DrawEngineVulkan::DoFlush() {
 			if (pipeline != lastPipeline_) {
 				vkCmdBindPipeline(cmd, VK_PIPELINE_BIND_POINT_GRAPHICS, pipeline->pipeline);
 				lastPipeline_ = pipeline;
+				if (lastPipeline_ && !lastPipeline_->useBlendConstant && pipeline->useBlendConstant) {
```
Comment (unknownbrackets) [inline GPU/Vulkan/DrawEngineVulkan.cpp]: Wait, those are equal now.  Doesn't this line need to be one higher - above `lastPipeline_ = pipeline;`?

-[Unknown]

File: GPU/Vulkan/DrawEngineVulkan.cpp [hunk 3]
```diff
@@ -986,8 +989,11 @@ void DrawEngineVulkan::DoFlush() {
 				if (pipeline != lastPipeline_) {
 					vkCmdBindPipeline(cmd, VK_PIPELINE_BIND_POINT_GRAPHICS, pipeline->pipeline);  // TODO: Avoid if same as last draw.
 					lastPipeline_ = pipeline;
+					if (lastPipeline_ && !lastPipeline_->useBlendConstant && pipeline->useBlendConstant) {
```
Comment (unknownbrackets) [inline GPU/Vulkan/DrawEngineVulkan.cpp]: Here too.

-[Unknown]

File: Windows/GPU/WindowsVulkanContext.cpp [hunk 4]
```diff
@@ -124,17 +124,7 @@ static VkBool32 VKAPI_CALL Vulkan_Dbg(VkDebugReportFlagsEXT msgFlags, VkDebugRep
...
-		return false;
-
-	// This seems like a bogus result when submitting two command buffers in one go, one creating the image, the other one using it.
-	if (msgCode == 6 && startsWith(pMsg, "Cannot submit cmd buffer using image"))
-		return false;
-	if (msgCode == 11)
-		return false;
-	// Silence "invalid reads of buffer data" - usually just uninitialized color buffers that will immediately get cleared due to our
-	// lacking clearing optimizations.
-	if (msgCode == 15 && objType == VK_DEBUG_REPORT_OBJECT_TYPE_DEVICE_MEMORY_EXT)
```
Comment (unknownbrackets) [inline Windows/GPU/WindowsVulkanContext.cpp]: Yay.

-[Unknown]

File: GPU/Vulkan/DrawEngineVulkan.cpp (hunk 2 above)
Comment (hrydgard) [reply GPU/Vulkan/DrawEngineVulkan.cpp]: Yes, good catch, thanks!

**Pull request merged**